e2e/cypress/support/ui/cloud_billing.js

```diff
@@ -7,3 +7,55 @@ Cypress.Commands.add('uiGetPaymentCardInput', () => {
         its('0.contentDocument.body').should('not.be.empty').
         then(cy.wrap);
 });
+
+Cypress.Commands.add('waitForResources', (resources = []) => {
+    const globalTimeout = 20000;
+    const resourceCheckInterval = 2000;
+    const idleTimesInit = 3;
+    let idleTimes = idleTimesInit;
+    let resourcesLengthPrevious;
+    let timeout;
+
+    return new Cypress.Promise((resolve, reject) => {
+        const checkIfResourcesLoaded = () => {
+            const resourcesLoaded = cy.state('window').
+                performance.getEntriesByType('resource').
+                filter((r) => !['script', 'xmlhttprequest', 'fetch'].includes(r.initiatorType));
+
+            const allFilesFound = resources.every(
+                (resource) => {
+                    const found = resourcesLoaded.filter(
+                        (resourceLoaded) => {
+                            return resourceLoaded.name.includes(resource.name);
+                        },
+                    );
+                    if (found.length === 0) {
+                        return false;
+                    }
+                    return !resource.number || found.length >= resource.number;
+                },
+            );
+
+            if (allFilesFound) {
+                if (resourcesLoaded.length === resourcesLengthPrevious) {
```

Comment:
I don't think I understand this logic. Why aren't we just resolving immediately once all the files are found?